### ISSUE TITLE
build prod failed

### ISSUE THREAD

**Issue by loyalpocket** [driftyco/ionic-storage#79]
Tried to do "ionic build android --prod --release" but failed:

Failed on type {"filePath":"E:/ionic/***projectname***/node_modules/@ionic/storage/es2015/storage.d.ts","name":"Storage","members":[]} with error Error: Error encountered resolving symbol values statically. Expression form not supported (position 102:31 in the original .ts file), resolving symbol Storage in E:/ionic/***projectname***/node_modules/@ionic/storage/es2015/storage.d.ts

build prod failed: Error encountered resolving symbol values statically. Expression form not supported
            (position 102:31 in the original .ts file), resolving symbol Storage in
            E:/ionic/***projectname***/node_modules/@ionic/storage/es2015/storage.d.ts

"@ionic/storage": "^1.1.7"

Any ideas?

**Comment by NoNameProvided** [comment]
You have to write a factory function:

```ts
export function provideStorage() { 
  return new Storage(); 
}
```

Then import it like this: 

```ts
{provide: Storage, useFactory: provideStorage},
```

**Comment by loyalpocket** [comment]
@NoNameProvided Thanks for your reply, but where to add the factory function and where to import?
In app.modules.ts, I have "import { Storage } from '@ionic/storage';", and "Storage" in providers.

**Comment by loyalpocket** [comment]
@NoNameProvided  Sorry, just saw this introduction:

Configuring Storage (new in 1.1.7)

The Storage engine can be configured both with specific storage engine priorities, or custom configuration options to pass to localForage. See the localForage config docs for possible options: https://github.com/localForage/localForage#configuration

import { Storage } from '@ionic/storage';

export function provideStorage() {
 return new Storage(['sqlite', 'websql', 'indexeddb'], { name: '__mydb' } /* optional config */);
}

@NgModule({
 declarations: ...,
 imports: ...,
 bootstrap: ...,
 entryComponents: ...,
  providers: [
    { provide: Storage, useFactory: provideStorage }
  ]
})
export class AppModule {}

Thanks for your help!

**Comment by jgw96** [comment]
Hello all, this has been fixed in the new 2.0.0 release of Ionic storage! https://github.com/driftyco/ionic-storage/releases/tag/v2.0.0
